crates/stackable-operator/src/commons/product_image_selection.rs

```diff
@@ -130,8 +130,14 @@ impl ProductImage {
                 let stackable_version = match &image_selection.stackable_version {
                     Some(stackable_version) => stackable_version,
                     None => {
-                        if operator_version.starts_with("0.0.0-") {
-                            "0.0.0-dev"
+                        if operator_version.starts_with("0.0.0-pr") {
+                            let override_version = "0.0.0-dev";
+                            tracing::warn!(
```

Comment:
The `warn` surprises me a bit (as everything works as intended - there are no pr versions of our product images), do you think an `info` would be sufficient?

Reply:
It's something that deviates from the expected behaviour in a way that should never happen during normal execution.

I'd want that called out _hard_ since it's likely to cause downstream confusion.

Reply:
> It's something that deviates from the expected behaviour

What would be expected behavior when using `0.0.0-pr812` (from this PR) here? I would expect `product:0.0.0-dev`

Reply:
I would expect that the product image version matches the operator version, since that's the behaviour we have at all other times. If I run `operator:1.2.3` I get `product:1.2.3`. I would expect that pattern to keep holding.

Going out on a limb, *you* only expect `0.0.0-dev` because you already knew about this special case.